src/componants/Article.jsx

```diff
@@ -0,0 +1,40 @@
+import { useContext, useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
+import { getArticleById } from "../../api"
+import categoryContext from "../contexts/categoryContexts"
+
+function Article (){
+  const {article_id} = useParams()
+  const [article, setArticle] = useState({})
+  const {setCategory} = useContext(categoryContext)
+
+  useEffect(() => {
+
```

Comment:
You could add some loading state here to conditionally render something to tell a user the data is still being fetched while they are waiting (for those on a slower internet connection) 💡